src/udev/udevadm-info.c

```diff
@@ -263,19 +310,19 @@ static void cleanup_db(void) {
 
         dir2 = opendir("/run/udev/links");
         if (dir2)
-                cleanup_dir(dir2, 0, 2);
+                cleanup_dirs_after_db_cleanup(dir2, dir1);
 
         dir3 = opendir("/run/udev/tags");
         if (dir3)
-                cleanup_dir(dir3, 0, 2);
+                cleanup_dirs_after_db_cleanup(dir3, dir1);
 
         dir4 = opendir("/run/udev/static_node-tags");
         if (dir4)
                 cleanup_dir(dir4, 0, 2);
 
         dir5 = opendir("/run/udev/watch");
         if (dir5)
-                cleanup_dir(dir5, 0, 1);
+                cleanup_dir_after_db_cleanup(dir5, dir1);
```

Comment:
This is not your fault, but recent udev also stores symlinks from watch handle to device id in 'watch' directory.
If you'd like, could you support such files?

If not, that's ok in this PR. I will do that later.

Reply:
It may be. I will take a look. Let's leave the code as is at least in this PR.